light-client/src/components/verifier.rs

```diff
@@ -78,47 +82,48 @@ impl ProdVerifier {
     }
 }
 
+impl Default for ProdVerifier {
+    fn default() -> Self {
+        Self::new(
+            ProdPredicates,
+            ProdVotingPowerCalculator,
+            ProdCommitValidator,
+            ProdHeaderHasher,
+        )
+    }
+}
+
 impl Verifier for ProdVerifier {
-    fn validate_light_block(
-        &self,
-        light_block: &LightBlock,
-        trusted_state: &TrustedState,
-        options: &Options,
-    ) -> Verdict {
-        crate::predicates::validate_light_block(
+    fn verify(&self, untrusted: &LightBlock, trusted: &TrustedState, options: &Options) -> Verdict {
+        let verdict: Verdict = preds::validate_light_block(
             &*self.predicates,
             &self.commit_validator,
             &self.header_hasher,
-            &trusted_state,
-            &light_block,
+            &trusted,
+            &untrusted,
             options,
         )
-        .into()
-    }
+        .into();
 
-    fn verify_overlap(
-        &self,
-        light_block: &LightBlock,
-        trusted_state: &TrustedState,
-        options: &Options,
-    ) -> Verdict {
-        crate::predicates::verify_overlap(
-            &*self.predicates,
-            &self.voting_power_calculator,
-            &trusted_state,
-            &light_block,
-            options,
-        )
-        .into()
-    }
+        let verdict = verdict.and_then(|| {
```

Comment:
If I understand correctly overlap verification should only be done if the two headers are non adjacent and in the case they are adjacent we should just check hashes as is done in `valid_next_validator_set`. While here `verify_overlap` is always checked disregarding adjacency relatinoship.

Reply:
Good catch, that's indeed an issue, which Ethan raised in #294. Glad to see you guys agree :)
Should be fixed in 953591a4296c32f4ebdd7b6158618634889d8ffa if you want to take another look.